host/platform/Linux/capture/XCB/src/xcb.c

```diff
@@ -228,13 +293,94 @@ static CaptureResult xcb_getFrame(FrameBuffer * frame,
   return CAPTURE_RESULT_OK;
 }
 
+static int pointerThread(void * unused)
+{
+  while (!this->stop)
+  {
+    if (this->stop)
+      break;
+
+    CapturePointer pointer = { 0 };
+
+    this->curC = xcb_xfixes_get_cursor_image_unchecked(this->xcb);
+
+    void * data;
+    uint32_t size;
+    if (!this->getPointerBufferFn(&data, &size))
+    {
+      DEBUG_WARN("Failed to get a pointer buffer");
+      continue;
+    }
+
+    xcb_xfixes_get_cursor_image_reply_t * curReply;
+    curReply = xcb_xfixes_get_cursor_image_reply(this->xcb, this->curC, NULL);
+    if (curReply == NULL)
+    {
+      DEBUG_WARN("Failed to get cursor reply");
+      continue;
+    }
+
+    if(curReply->xhot != this->mouseHotX || curReply->yhot != this->mouseHotY)
+    {
+      pointer.shapeUpdate = true;
+      this->mouseHotX = curReply->xhot;
+      this->mouseHotY = curReply->yhot;
+
+      uint32_t * src = xcb_xfixes_get_cursor_image_cursor_image(curReply);
+      uint32_t * dst = (uint32_t *) (data);
+      memcpy(dst, src, curReply->width * curReply->height * sizeof(uint32_t));
+    }
+    else
+    {
+      pointer.shapeUpdate = false;
+    }
+
+    if(curReply->x != this->mouseX || curReply->y != this->mouseY)
+    {
+      pointer.positionUpdate = true;
+      this->mouseX = curReply->x;
+      this->mouseY = curReply->y;
+    }
+    else
+    {
+      pointer.positionUpdate = false;
+    }
+
+    if(!pointer.positionUpdate && !pointer.shapeUpdate)
+    {
+      free(curReply);
+      nsleep(1000);
```

Comment:
Why not get rid of the `continue` and move the `postPointerBufferFn` call into the `if` with logic toggled? This avoids duplicating the cleanup code.